Assets/Scripts/Game/Questing/Foe.cs

```diff
@@ -337,9 +366,19 @@ public override void RestoreSaveData(object dataIn)
             if (dataIn == null)
                 return;
 
-            SaveData_v1 data = (SaveData_v1)dataIn;
+            SaveData_v2 data;
+
+            if (dataIn is SaveData_v1)
```

Comment:
I had expected the `[fsObject("v2", typeof(SaveData_v1))]` attribute on the struct to cause fsSerializer to migrate the struct automatically. This is not the case. I suspect the migration only happens if `Foe.SaveData_v2` was the top-level struct we were deserialized. Here, the save data is nested as a generic object in `ResourceSaveData_v1.resourceSpecific`.

So, I do the migration manually here